### PR TITLE
feat: Automagically load experiences on demand

### LINKED ISSUE #9: Find a way to automatically call ReactExperienceLoader.load()
While being able to explicitly call `ReactExperienceLoader.load()` at strategic points during runtime is nice, it would be a nice enhancement if `react-native-lazy-index` could make those calls by itself in certain scenarios. For instance, could we leverage `AppRegistry.registerComponent()` (and `BatchedBridge.registerCallableModule()`) to determine that we should call `ReactExperienceLoader.load()` before the app is loaded?

### PR DESCRIPTION
**Pull request by tido64**
### BREAKING CHANGE

`ReactExperienceLoader.load()` has been removed. Modules registered using `AppRegistry` or `BatchedBridge` are now
automatically loaded on demand. The configuration schema has also been changed to reflect this.

Resolves #9.

### CODE REVIEW

File: generateLazyIndex.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        lazyIndex.push(
+          `AppRegistry.registerComponent("${appKey.value}", () => {`,
+          `  // Source: ${path.relative(process.cwd(), modulePath)}`,
+          `  require("${state.module}");`,
+          `  return AppRegistry.getRunnable("${appKey.value}").componentProvider();`,
+          `});`
+        );
+      } else if (
+        object.name === "BatchedBridge" &&
+        property.name === "registerCallableModule"
```
Comment (tido64) [inline generateLazyIndex.js:178]: Should probably also wrap `BatchedBridge.registerLazyCallableModule()`.

**Review by sweggersen — APPROVED** [review]
LGTM

File: generateLazyIndex.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+ * @param {number} defaultValue
+ */
+function parseIntDefault(s, defaultValue) {
+  const value = parseInt(s || "");
+  return isNaN(value) ? defaultValue : value;
+}
+
+/**
+ * @param {string} moduleId
+ * @param {string=} parentModule
```
[the remark this answers is not in the document]
Comment (tido64) [reply generateLazyIndex.js:47]: The `=` is to make the parameter optional. We are not passing `parentModule` when reading from `package.json`.

**Pull request merged**